sdv/datasets/demo.py

```diff
@@ -0,0 +1,123 @@
+"""Methods to load datasets."""
+
+import io
+import logging
+import os
+import shutil
+import tempfile
+import urllib.request
+from zipfile import ZipFile
+
+import pandas as pd
+
+from sdv.metadata.multi_table import MultiTableMetadata
+from sdv.metadata.single_table import SingleTableMetadata
+
+LOGGER = logging.getLogger(__name__)
+BUCKET_URL = 'https://sdv-demo-datasets.s3.amazonaws.com'
+METADATA_FILENAME = 'metadata.json'
+
+
+def _get_dataset_url(modality, dataset_name):
+    return os.path.join(BUCKET_URL, modality.upper(), dataset_name + '.zip')
+
+
+def _validate_args_download_demo(modality, output_folder_name):
+    possible_modalities = ['single_table', 'multi_table', 'sequential']
+    if modality not in possible_modalities:
+        raise ValueError(f"'modality' must be in {possible_modalities}.")
+
+    if output_folder_name and os.path.exists(output_folder_name):
+        raise ValueError(
+            f"Folder '{output_folder_name}' already exists. Please specify a different name "
+            "or use 'load_csvs' to load from an existing folder."
+        )
+
+
+def _download(modality, dataset_name, output_folder_name):
+    dataset_url = _get_dataset_url(modality, dataset_name)
+    LOGGER.info(f'Downloading dataset {dataset_name} from {dataset_url}')
+    try:
+        response = urllib.request.urlopen(dataset_url)
+    except urllib.error.HTTPError:
+        raise ValueError(
+            f"Invalid dataset name '{dataset_name}'. "
+            'Make sure you have the correct modality for the dataset name or '
+            "use 'get_available_demos' to get a list of demo datasets."
+        )
+
+    LOGGER.info(f'Extracting dataset into {output_folder_name}')
+    bytes_io = io.BytesIO(response.read())
+    with ZipFile(bytes_io) as zf:
+        os.makedirs(output_folder_name, exist_ok=True)
+        zf.extractall(output_folder_name)
+        os.remove(os.path.join(output_folder_name, 'metadata_v0.json'))
+        os.rename(
+            os.path.join(output_folder_name, 'metadata_v1.json'),
+            os.path.join(output_folder_name, METADATA_FILENAME)
```

Comment:
would it be possible to extract the objects directly in memory instead of storing them to disk in a temp file? something like what's discussed here? https://stackoverflow.com/questions/5710867/downloading-and-unzipping-a-zip-file-without-writing-to-disk
And then we'd only save to disk if requested by the user

Reply:
I thought about it, but the code would be a little more complicated (I need to pass a path to `load_from_json`) and a little less readable, while also potentially increasing memory usage (since, if you don't do something fancy, you have a copy of the zip file and the object in memory at the same time). It just doesn't seem worth it.

Reply:
As @amontanez24 pointed, it is possible to read the object in memory and once you exit the `with` indentation the garbage collector will close the `zip` file and unload it from memory, only the assigned variable in it will stay.

I'm not sure about the format that the `json` will come out, but if it's a `dict` just use the private method `_load_from_dict`:  https://github.com/sdv-dev/SDV/blob/V1.0.0.dev/sdv/metadata/single_table.py#L512